utilities/release.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+import sys
+from sh import git, ErrorReturnCode
+from semver import VersionInfo
+import argparse
+
+
+class Repository:
+    """
+    Encapsulates some logic for running git commands within the repository,
+    particularly related to creating release tags/branches.
+    """
+    def __init__(self, root='.', dry_run=False):
+        self.root = root
+        self.dry_run = dry_run
+        self.git = git.bake('--no-pager', C=root)
```

Comment:
This is sort of a nitpick, but maybe we can have the class save the current branch in the repo and check it out in the destructor. It's a minor thing but I personally don't like scripts that move things around and dump me somewhere different than where I was 🙂

Reply:
Excellent point - I'll definitely put something like that in. Perhaps making it a context manager would be neat?